src/main/java/teammates/common/datatransfer/attributes/InstructorAttributes.java

```diff
@@ -18,131 +18,49 @@
     public static final String DEFAULT_DISPLAY_NAME = "Instructor";
 
     // Note: be careful when changing these variables as their names are used in *.json files.
+
+    /** Required fields. */
     public String googleId;
     public String courseId;
     public String name;
     public String email;
-    public Boolean isArchived;
+
+    /** Optional fields. */
     public String key;
     public String role;
-    public boolean isDisplayedToStudents;
     public String displayedName;
-
+    public Boolean isArchived;
+    public boolean isDisplayedToStudents;
     public InstructorPrivileges privileges;
 
-    /**
-     * Creates a new instructor with default access level and default displayedName.
-     *
-     * @deprecated only to be used for testing
-     */
-    @Deprecated
-    public InstructorAttributes(String googleId, String courseId, String name, String email) {
-        this(googleId, courseId, name, email,
-             Const.InstructorPermissionRoleNames.INSTRUCTOR_PERMISSION_ROLE_COOWNER, DEFAULT_DISPLAY_NAME,
-             new InstructorPrivileges(Const.InstructorPermissionRoleNames.INSTRUCTOR_PERMISSION_ROLE_COOWNER));
-    }
-
-    /**
-     * Creates a new instructor with params specified (isDisplayedToStudent is set to true by default).
-     */
-    public InstructorAttributes(String googleId, String courseId, String name, String email, String role,
-                                String displayedName, String instructorPrivilegesAsText) {
-        this.googleId = SanitizationHelper.sanitizeGoogleId(googleId);
-        this.courseId = SanitizationHelper.sanitizeTitle(courseId);
-        this.isArchived = false;
-        this.name = SanitizationHelper.sanitizeName(name);
-        this.email = email;
-        this.role = SanitizationHelper.sanitizeName(role);
-        this.isDisplayedToStudents = true;
-        this.displayedName = SanitizationHelper.sanitizeName(displayedName);
-        this.privileges = getInstructorPrivilegesFromText(instructorPrivilegesAsText);
-    }
 
-    /**
-     * Creates an instructor (isDisplayedToStudent is set to true by default).
-     */
-    public InstructorAttributes(String googleId, String courseId, String name, String email, String role,
-                                String displayedName, InstructorPrivileges privileges) {
-        this.googleId = SanitizationHelper.sanitizeGoogleId(googleId);
-        this.courseId = courseId;
-        this.isArchived = false;
-        this.name = SanitizationHelper.sanitizeName(name);
-        this.email = email;
-        this.role = SanitizationHelper.sanitizeName(role);
-        this.isDisplayedToStudents = true;
-        this.displayedName = SanitizationHelper.sanitizeName(displayedName);
-        this.privileges = privileges;
+    /** Return new builder instance. */
+    public static Builder builder(String googleId, String courseId, String name, String email) {
```

Comment:
Add more descriptions to the javadoc
e.g. To say that `isDisplayedToStudent` is set to true by default